filetracker/servers/files.py

```diff
@@ -93,49 +85,64 @@ def _file_headers(self, path):
             ]
 
     def handle_GET(self, environ, start_response):
-        # GET also supports /list endpoint, not just /files like other methods.
-        if environ['PATH_INFO'].startswith('/list/'):
+        endpoint, path = base.get_endpoint_and_path(environ)
+        if endpoint == 'list':
             return self.handle_list(environ, start_response)
+        elif endpoint == 'files':
+            path = os.path.join(self.dir, path)
 
-        path = os.path.join(self.dir, self.get_path(environ))
+            if not os.path.isfile(path):
+                raise base.HttpError(
+                        '404 Not Found', 'File "{}" not found'.format(path))
 
-        if not os.path.isfile(path):
-            start_response('404 Not Found', [('Content-Type', 'text/plain')])
-            return [('File not found: %s' % path).encode()]
-        start_response('200 OK', self._file_headers(path))
-        return self._fileobj_iterator(open(path, 'rb'))
+            start_response('200 OK', self._file_headers(path))
+            return self._fileobj_iterator(open(path, 'rb'))
+        else:
+            raise base.HttpError(
+                    '400 Bad Request',
+                    'Unknown endpoint "{}", expected "files" or "list"'
+                    .format(endpoint))
 
     def handle_HEAD(self, environ, start_response):
-        path = os.path.join(self.dir, self.get_path(environ))
+        endpoint, path = base.get_endpoint_and_path(environ)
+        if endpoint != 'files':
+            raise base.HttpError('400 Bad Request',
+                                 'HEAD can be only performed on "/files/..."')
```

Comment:
I think this breaks HTTP specification: https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.4
HEAD SHOULD return the same headers as its respective GET, which is now not the case for /list/...